server/sonar-server/src/main/java/org/sonar/server/es/textsearch/ComponentTextSearchFeatureRepertoire.java

```diff
@@ -0,0 +1,151 @@
+        .boost(2f);
+    }
+  },
+  PARTIAL(GENERATE_RESULTS) {
+    @Override
+    public QueryBuilder getQuery(ComponentTextSearchQuery query) {
+      BoolQueryBuilder queryBuilder = boolQuery();
+      split(query.getQueryText())
+        .map(text -> tokenQuery(text, query.getFieldName(), SEARCH_GRAMS_ANALYZER))
+        .forEach(queryBuilder::must);
+      return queryBuilder
+        .boost(0.5f);
+    }
+  },
+  KEY(GENERATE_RESULTS) {
+    @Override
+    public QueryBuilder getQuery(ComponentTextSearchQuery query) {
+      return matchQuery(SORTABLE_ANALYZER.subField(query.getFieldKey()), query.getQueryText())
+        .boost(50f);
+    }
+  },
+  RECENTLY_BROWSED(CHANGE_ORDER_OF_RESULTS) {
+    @Override
+    public Stream<QueryBuilder> getQueries(ComponentTextSearchQuery query) {
+      Set<String> recentlyBrowsedKeys = query.getRecentlyBrowsedKeys();
+      if (recentlyBrowsedKeys.isEmpty()) {
+        return Stream.empty();
+      }
+      return Stream.of(termsQuery(query.getFieldKey(), recentlyBrowsedKeys).boost(100f));
+    }
+  },
+  FAVORITE(CHANGE_ORDER_OF_RESULTS) {
+    @Override
+    public Stream<QueryBuilder> getQueries(ComponentTextSearchQuery query) {
+      Set<String> favoriteKeys = query.getFavoriteKeys();
+      if (favoriteKeys.isEmpty()) {
+        return Stream.empty();
+      }
+      return Stream.of(termsQuery(query.getFieldKey(), favoriteKeys).boost(1000f));
+    }
+  };
+
+  private final UseCase useCase;
+
+  ComponentTextSearchFeatureRepertoire(UseCase useCase) {
+    this.useCase = useCase;
+  }
+
+  public QueryBuilder getQuery(ComponentTextSearchQuery query) {
+    throw new UnsupportedOperationException();
+  }
+
+  public static Stream<String> split(String queryText) {
+    return Arrays.stream(
+      queryText.split(DefaultIndexSettings.SEARCH_TERM_TOKENIZER_PATTERN))
+      .filter(StringUtils::isNotEmpty);
+  }
+
+  protected BoolQueryBuilder prefixAndPartialQuery(String queryText, String originalFieldName, DefaultIndexSettingsElement analyzer) {
+    BoolQueryBuilder queryBuilder = boolQuery();
+
+    AtomicBoolean first = new AtomicBoolean(true);
+    split(queryText)
+      .map(queryTerm -> {
+
+        if (first.getAndSet(false)) {
+          return tokenQuery(queryTerm, originalFieldName, analyzer);
+        }
+
+        return tokenQuery(queryTerm, originalFieldName, SEARCH_GRAMS_ANALYZER);
+      })
+      .forEach(queryBuilder::must);
+    return queryBuilder;
+  }
+
+  protected MatchQueryBuilder tokenQuery(String queryTerm, String fieldName, DefaultIndexSettingsElement analyzer) {
+    // We will truncate the search to the maximum length of nGrams in the index.
+    // Otherwise the search would for sure not find any results.
+    String truncatedQuery = StringUtils.left(queryTerm, DefaultIndexSettings.MAXIMUM_NGRAM_LENGTH);
+    return matchQuery(analyzer.subField(fieldName), truncatedQuery);
+  }
+
+  public UseCase getUseCase() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the "@Override" annotation above this method signature [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1161)